reactive/jenkins.py

```diff
@@ -173,6 +180,49 @@ def migrate_charm_data():
     configuration.migrate()
 
 
+@when('nrpe-external-master.available')
+def update_nrpe_config(nagios):
+    udata = unitdata.kv()
+    nagios_hostname = udata.get('nagios.hostname', None)
+    nagios_host_context = udata.get('nagios.host_context', None)
+
+    # require the nrpe-external-master relation to provide the host context
+    if in_relation_hook() and relation_id().\
+            startswith('nrpe-external-master:'):
+        rel = relation_get()
+        if 'nagios_host_context' in rel:
+            nagios_host_context = rel['nagios_host_context']
+            udata.set('nagios.host_context', nagios_host_context)
+
+            # We have to strip the nagios host context from the nagios hostname
+            # since the nagios.add_check will put it back again...
+            nagios_hostname = rel['nagios_hostname']
+            if nagios_hostname.startswith(nagios_host_context + '-'):
+                nagios_hostname = nagios_hostname[len(nagios_host_context +
+                                                      '-'):]
+
+            udata.set('nagios.hostname', nagios_hostname)
+
+    if not nagios_hostname or not nagios_host_context:
+        return
```

Comment:
I do think this would be better put into the interface layer as it is largely boilerplate, really I think nagios.add_check should just automatically fill in the context and unit. Unfortunately that is not how the interface was written nor how it has been historically used and that is not the only problem with how nrpe is often used in deployments. Fixing it is no small task and is out of scope for the work I am doing now so I put in that boilerplate code.

Typically each deployed charm has a config option for nagios_context. This means that at deploy time you are setting the host context over and over for each charm related to nrpe (usually everything). The boiler plate code I added works around that problem by pulling the context from the relation so it only needs to be set for the nrpe charm. If we pull the boiler plate code out then the alternative is to add in the config option for the context.